force-app/main/default/lwc/donationHistoryTable/donationHistoryTable.js

```diff
@@ -0,0 +1,44 @@
+import { LightningElement } from 'lwc';
+import {MOCK_DATA} from './donationHistoryTableData';
+//TODO: set custom label for datatable aria label
+
+const columns = [
+    { label: 'Date', fieldName: 'date', type: 'date', typeAttributes:{
+        year: "numeric",
+        month: "short",
+        day: "2-digit"
+    },
+    cellAttributes: { alignment: 'right' }},
+    { label: 'Donor', fieldName: 'donor', type: 'text' },
+    { label: 'Amount', fieldName: 'amount', type: 'currency', },
+    { label: 'Payment Method', fieldName: 'paymentMethod', type: 'text', },
+];
+
+
+export default class DonationHistoryTable extends LightningElement {
+    data = [];
+    columns = columns;
+    totalNumberOfRows = MOCK_DATA.length;
+    
+    // eslint-disable-next-line @lwc/lwc/no-async-await
+    async connectedCallback() {
+        const data = MOCK_DATA.slice(0, 50);
+        this.data = data;
+    }
+
+    loadMoreDonationData(event){
+        //TODO: add wired service
+        //Display a spinner to signal that data is being loaded
+        event.target.isLoading = true;
+        if (this.data.length >= this.totalNumberOfRows) {
+            event.target.enableInfiniteLoading = false;
+            this.loadMoreStatus = 'No more data to load';
```

Comment:
use a constant or a custom label(if it is visible to the customer)

Reply:
this is only for debugging purposes